bucket4j-spring-boot-starter-context/src/main/java/com/giffing/bucket4j/spring/boot/starter/context/constraintvalidations/DurationChronoUnitValidator.java

```diff
@@ -0,0 +1,14 @@
+package com.giffing.bucket4j.spring.boot.starter.context.constraintvalidations;
+
+import java.time.temporal.ChronoUnit;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class DurationChronoUnitValidator implements ConstraintValidator<ValidDurationChronoUnit, ChronoUnit> {
+
+	@Override
+	public boolean isValid(ChronoUnit value, ConstraintValidatorContext context) {
+		return value != null && (value == ChronoUnit.DAYS || !value.isDurationEstimated());
```

Comment:
I'm trying to understand the check but I didn't get it. Can you please explain the part:
(value == ChronoUnit.DAYS || !value.isDurationEstimated()).

The method isDurationEstimated check if it equals DAYS:
return this.compareTo(DAYS) >= 0;

Reply:
The Duration.of() method which is used when creating rate limits, does not allow estimated timeunits. The 'DAYS' unit however is an exception to this rule. According to the "isDurationEstimated" method a day is an estimated unit because of daylight saving. The Duration.of() method on the other hand has a specific check that treats days as 24hours, making it no longer estimated. For this reason this specific check is added so Days are allowed. 

Good that you mention your confusion though, I will add a comment above the code to explain the behavior for future developers that look into it.